### PR TITLE
Add NNCF quantization for ImageBind notebook

### PR DESCRIPTION
**Pull request by l-bat**

### CODE REVIEW

File: notebooks/239-image-bind/README.md
Comment (MaximProshin) [inline notebooks/239-image-bind/README.md:13]: I propose to align it everywhere with OV documentation where we agreed to have the convert step (MO+FEs) and the optimize step (NNCF). I.e. 1st notebook is about the conversion to IR and 2nd is about the optimization by 8-bit quantization.

File: notebooks/239-image-bind/README.md
Comment (MaximProshin) [inline notebooks/239-image-bind/README.md:18]: I would clearly state "1st notebook"

File: notebooks/239-image-bind/README.md
Comment (MaximProshin) [inline notebooks/239-image-bind/README.md:27]: I would clearly state "2nd notebook"

File: notebooks/239-image-bind/README.md
Comment (MaximProshin) [inline notebooks/239-image-bind/README.md:25]: Let's move this paragraph after the content description of the 2nd notebook

File: notebooks/239-image-bind/README.md
Comment (MaximProshin) [inline notebooks/239-image-bind/README.md:29]: with NNCF Tool API in OpenVINO => API of NNCF

File: notebooks/239-image-bind/README.md
Comment (MaximProshin) [inline notebooks/239-image-bind/README.md:31]: can we also add model size?

File: notebooks/239-image-bind/README.md
Comment (eaidova) [reply notebooks/239-image-bind/README.md]: probably if we will agreed to upload model somewhere and it will be applicable (not image bind case), third simple notebook for demo with opportunity to select model (fp/int8) can take place if we speak about general structure

File: notebooks/239-image-bind/README.md
Comment (eaidova) [reply notebooks/239-image-bind/README.md]: model size comparision added in readme but missed in table of content of notebook, could you please align?

**Comment by eaidova** [comment]
@l-bat   I think it is also will be nice to add link on second notebook in first notebook like section of "next steps" to make them more connected.

Also in some cases, when quantization resource consuming process may we add some notes about the recommended amount of RAM and maybe the approximate time of quantization?

I correctly understand that by default nncf quantization will be performed with openvino backend? 
what if several approaches can be applicable (e.g. we want to compare ov quantization and pytorch) or such situations are not desired and we should demonstrate only the most convenient and optimal way? (not related to this notebook, I just trying to clarify whole range of rules for contribution guide)

File: notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb
Comment (eaidova) [inline notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #4.    
</pre><p>it maybe strange situation when you have no repo dir, but trying to get IR inside in cell below. Probably in this case the first notebook should executed before inside this or at least provide some error/warning that model is not converted yet? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1155/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (eaidova) [inline notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:1]: <p>can you please also provide links on documentation page or/and nncf repo?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1155/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (eaidova) [inline notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:1]: <p>please provide link on dataset page or paper</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1155/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='6'/>

Comment (eaidova) [inline notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #16.    def laion2B_preprocess_train(example, image_column="url", text_column="caption"):
</pre><p>name of preprocessing looks confused, can we rename it to laion_preprocess? (in description below you mention lion-400m, but function claims laion-2b and train suffix also may lead to confusion as we do not perform QAT, right?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1155/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='7'/>

Comment (eaidova) [inline notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:1]: <pre class="ql-syntax" spellcheck="false">Line #1.    ov_quantized_modality_models = {
</pre><p>it is recommended to create core object once, then use the same object for all models compilation or if you do not need core at all, please use function instead:</p><p>from openvino.runtime import compile_model </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1155/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='25'/>

File: notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb
Comment (MaximProshin) [inline notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:18]: I would add a reference to 1st notebook

File: notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb
Comment (MaximProshin) [inline notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:119]: can you add a short description before each function about its goal?

File: notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb
Comment (MaximProshin) [inline notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:173]: I get an error in this cell (on Windows): 

_ModuleNotFoundError                       Traceback (most recent call last)
Cell In[4], line 1
----> 1 from datasets import load_dataset
      2 import itertools
      3 import torch

ModuleNotFoundError: No module named 'datasets'_

Adding !pip install datasets directly to the notebook helped.

File: notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb
Comment (MaximProshin) [inline notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:273]: NNCF logs are too long and even confusing (ignoring only reduce layer, but many more layers are ignored due to Transformer model type). I suggest to disable "debug" at least.

Not a stopper for this notebook but we should decide what to do in general.

File: notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb
Comment (MaximProshin) [reply notebooks/239-image-bind/239-image-bind-quantize-nncf.ipynb:173]: Also quantization of the last model failed with
ImportError: To support decoding audio files, please install 'librosa' and 'soundfile'.

**Review by eaidova — APPROVED** [review]

File: notebooks/239-image-bind/README.md
Comment (eaidova) [inline notebooks/239-image-bind/README.md:25]: can we add some note to highlight that the second part depends on first and requires to run first notebook?

**Review by MaximProshin — APPROVED** [review]
LGTM, after recent changes it works fine. I'm ok to merge it

File: notebooks/239-image-bind/239-image-bind-quantize.ipynb
Comment (eaidova) [inline notebooks/239-image-bind/239-image-bind-quantize.ipynb:1]: <p>on some platforms, there can be difficult to run librosa without soundfile as backend. Could you please add it in requiremnts?</p><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/1155/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (l-bat) [reply notebooks/239-image-bind/239-image-bind-quantize.ipynb:1]: Yes, I can. I didn't do it before, because soundfile is installed in convert notebook

**Pull request merged**
